pulp_docker/tests/functional/api/test_recursive_remove.py

```diff
@@ -0,0 +1,333 @@
+        ))['results'][0]['_href']
+        ml_iii_tag = self.client.get("{unit_path}?{filters}".format(
+            unit_path=DOCKER_TAG_PATH,
+            filters="name=ml_iii&{v_filter}".format(v_filter=self.latest_from_version),
+        ))['results'][0]['_href']
+        ml_iv_tag = self.client.get("{unit_path}?{filters}".format(
+            unit_path=DOCKER_TAG_PATH,
+            filters="name=ml_iv&{v_filter}".format(v_filter=self.latest_from_version),
+        ))['results'][0]['_href']
+        self.client.post(
+            DOCKER_RECURSIVE_ADD_PATH,
+            {
+                'repository': self.to_repo['_href'],
+                'content_units': [ml_i_tag, ml_ii_tag, ml_iii_tag, ml_iv_tag]
+            }
+        )
+        latest_version_href = self.client.get(self.to_repo['_href'])['_latest_version_href']
+        latest = self.client.get(latest_version_href)
+
+        self.assertEqual(latest['content_summary']['added']['docker.tag']['count'], 4)
+        self.assertEqual(latest['content_summary']['added']['docker.manifest']['count'], 9)
+        self.assertEqual(latest['content_summary']['added']['docker.blob']['count'], 10)
+
+        self.client.post(
+            DOCKER_RECURSIVE_REMOVE_PATH,
+            {
+                'repository': self.to_repo['_href'],
+                'content_units': [ml_i_tag, ml_ii_tag, ml_iii_tag, ml_iv_tag]
+            }
+        )
+        latest_version_href = self.client.get(self.to_repo['_href'])['_latest_version_href']
+        latest = self.client.get(latest_version_href)
+
+        self.assertEqual(latest['content_summary']['removed']['docker.tag']['count'], 4)
+        self.assertEqual(latest['content_summary']['removed']['docker.manifest']['count'], 9)
+        self.assertEqual(latest['content_summary']['removed']['docker.blob']['count'], 10)
+
+    def test_cannot_remove_tagged_manifest(self):
+        """
+        Try to remove a manifest (without removing tag). Creates a new version, but nothing removed.
+        """
+        manifest_a_tag = self.client.get("{unit_path}?{filters}".format(
+            unit_path=DOCKER_TAG_PATH,
+            filters="name=manifest_a&{v_filter}".format(v_filter=self.latest_from_version),
+        ))['results'][0]
+        self.client.post(
+            DOCKER_RECURSIVE_ADD_PATH,
+            {
+                'repository': self.to_repo['_href'],
+                'content_units': [manifest_a_tag['_href']]
+            }
+        )
+        latest_version_href = self.client.get(self.to_repo['_href'])['_latest_version_href']
+        latest = self.client.get(latest_version_href)
+        self.assertEqual(latest['content_summary']['added']['docker.tag']['count'], 1)
+        self.assertEqual(latest['content_summary']['added']['docker.manifest']['count'], 1)
+        self.assertEqual(latest['content_summary']['added']['docker.blob']['count'], 2)
+
+        self.client.post(
```

Comment:
@gmbnomis, this test covers the case you were interested in. The setup occurs differently (we recursive-add a tag rather than recursive-add a manifest and then tag it) but the state is the same at this point, which is where the actual test begins.

Reply:
@asmacdo I meant a different test. It could look like this:

1. Add T -> M -> B1, B2 recursively
2. Delete T non recursively, new repo version content is M -> B1,B2
3. Delete T recursively (T mus still be part of the global content pool, since it it part of a previous repo version)
4. Verify that M -> B1,B2 still exist in the created repo version

The test is meant to ensure that content outside of the repo version does not affect the remove operation at all (even when the outside content contains references to inside content).